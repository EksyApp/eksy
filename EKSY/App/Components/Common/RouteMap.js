import React, {Component} from 'react'
import MapView from 'react-native-maps'
import {MarkersShape, RegionShape, StyleShape} from '../../Utils/PropTypeShapes'

export default class RouteMap extends Component {
	
	
	render() {
		return (
				<MapView
						style={this.props.style}
						initialRegion={{
							...this.props.initialRegion,
							latitude: this.props.markers[0] ? this.props.markers[0].latitude : this.props.initialRegion.latitude,
							longitude: this.props.markers[0] ? this.props.markers[0].longitude : this.props.initialRegion.longitude
						}}
				>
					{this.props.markers.map((marker, index) => {
						return(
								<MapView.Marker key={index} coordinate={{latitude: marker.latitude, longitude: marker.longitude}}/>
						)
					})}
					
					<MapView.Polyline
							coordinates={this.props.markers.map((marker) => {
								return {latitude: marker.latitude, longitude: marker.longitude}
							})}
					/>
				</MapView>
		)
	}
}

RouteMap.propTypes = {
	style: StyleShape ,
	initialRegion: RegionShape,
	markers: MarkersShape
}