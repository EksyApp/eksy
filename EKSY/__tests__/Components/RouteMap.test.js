import React from 'react';
import renderer from 'react-test-renderer';
import RouteMap from "../../App/Components/Common/RouteMap";


test('RouteMap renders correctly', () => {
	const tree = renderer.create(
			<RouteMap
					initialRegion = {{latitude: 1, longitude: 1, latitudeDelta: 1, longitudeDelta: 1}}
					markers = {[{latitude: 1, longitude: 1}]}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});