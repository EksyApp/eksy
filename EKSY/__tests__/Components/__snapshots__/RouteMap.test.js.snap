// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteMap renders correctly 1`] = `
<MapView
  initialRegion={
    Object {
      "latitude": 1,
      "latitudeDelta": 1,
      "longitude": 1,
      "longitudeDelta": 1,
    }
  }
  style={undefined}
>
  <Marker
    coordinate={
      Object {
        "latitude": 1,
        "longitude": 1,
      }
    }
  />
  <Polyline
    coordinates={
      Array [
        Object {
          "latitude": 1,
          "longitude": 1,
        },
      ]
    }
  />
</MapView>
`;
